File: src/types/company.ts
export interface CompanyFinancials {
  code: string;
  name: string;
  fiscalYear: number | null;
  reportType: string | null;
  reportLabel: string | null;

  revenue: number | null;
  operatingIncome: number | null;
  netIncome: number | null;
  totalAssets: number | null;
  totalLiabilities: number | null;
  totalEquity: number | null;

  eps: number | null;
  roe: number | null;
  per: number | null;
  pbr: number | null;
}

File: backend/models/company.py
from pydantic import BaseModel


class CompanyFinancials(BaseModel):
    code: str
    name: str
    fiscalYear: int | None = None
    reportType: str | None = None  # "CFS" | "OFS"
    reportLabel: str | None = None

    revenue: float | None = None
    operatingIncome: float | None = None
    netIncome: float | None = None
    totalAssets: float | None = None
    totalLiabilities: float | None = None
    totalEquity: float | None = None

    eps: float | None = None
    roe: float | None = None
    per: float | None = None
    pbr: float | None = None
